lib/msf/core/exploit/remote/kerberos/client/base.rb

```diff
@@ -40,6 +40,12 @@ def build_server_name(opts = {})
                 name_string: name.split('/')
               )
             end
+
+            # @param [Rex::Proto::Kerberos::Model::KdcResponse] asrep The krb5 asrep response
+            # @return [String] A valid string format which can be cracked offline
+            def format_asrep_to_john_hash(asrep)
+              "$krb5asrep$#{asrep.enc_part.etype}$#{asrep.cname.name_string.join('/')}@#{asrep.ticket.realm}:#{asrep.enc_part.cipher[0...16].unpack1('H*')}$#{asrep.enc_part.cipher[16..].unpack1('H*')}"
+            end
```

Comment:
If we consider the existing functions in this file, I'm not sure if this belongs here 👀

Reply:
I've moved it to `as_response.rb`, thinking that is a better fit. Let me know if there is a better place for it 😅